File: 01_Array_Hashings/03_Two_Sum/01_Two_Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Dictionary<Integer, Integer> seen = new Hashtable<>();

        for (int i=0; i<nums.length; i++){
            int complement = target - nums[i];
            if (seen.get(complement) != null){
                return new int[] {seen.get(complement), i};
            }
            seen.put(nums[i], i);

        }

        return(new int[] {});
    }
}

File: 01_Array_Hashings/03_Two_Sum/01_Two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = dict()

        for i, v in enumerate(nums):
            print(i, v)
            complement = target - v
            if complement in seen:
                return [seen[complement], i]
            seen[v] = i

        return []